### DISCUSSION TITLE
Incorrect ISO 3166-2 format of state codes

### DISCUSSION THREAD

**Discussion by jigar-dhulla** [dr5hn/countries-states-cities-database#605] [Ideas]
Correct ISO format: `<CountryCode>-<Code>`. E.g. `IN-MH`

Format Stored: `<Code>`. E.g. `MH`

Correct format is referenced from here:
https://www.iso.org/obp/ui/#iso:code:3166:IN
https://www.iso.org/obp/ui/#iso:code:3166:GB
https://www.iso.org/obp/ui/#iso:code:3166:UG

When the State Code is Numeric **it may** conflict with the Primary ID when **searching for a state by ID or Code**.
For e.g. Uganda: https://www.iso.org/obp/ui/#iso:code:3166:UG

**Comment by dr5hn** [comment]
@jigarakatidus Thanks for reporting. 🎉 

I agree with you, Will fix them ASAP.

**Comment by jigar-dhulla** [comment]
> @jigarakatidus Thanks for reporting. 🎉
> 
> I agree with you, Will fix them ASAP.

This might be breaking change for those consuming APIs, so take care.  :)

Keep up the good work. Thanks.

**Comment by dr5hn** [comment]
@jigarakatidus Thanks for the heads up!

Indeed a breaking change, but I was thinking of just adding another column of data for this. For those who want to consume this way.

**Comment by jigar-dhulla** [comment]
> @jigarakatidus Thanks for the heads up!
> 
> Indeed a breaking change, but I was thinking of just adding another column of data for this. For those who want to consume this way.

That also works. But I would suggest to keep file size low and avoid duplication as this is getting distributed. :)

Currently, we are consuming the data only from CSV. So before adding to our DB, we just concatenate CountryCode-StateCode. This works for us. This issue is not burning in a way.

So probably, a newer **major** version can be released with a breaking change. Just to be in sync with the standard.
